.evergreen/config.yml

```diff
@@ -0,0 +1,99 @@
+# When a task that used to pass starts to fail
+# Go through all versions that may have been skipped to detect
+# when the task started failing
+stepback: true
+
+# Mark a failure as a system/bootstrap failure (purple box) rather then a task
+# failure by default.
+# Actual testing tasks are marked with `type: test`
+command_type: system
+
+# Protect ourself against rogue test case, or curl gone wild, that runs forever
+# 12 minutes is the longest we'll ever run
+exec_timeout_secs: 3600 # 12 minutes is the longest we'll ever run
+
+# What to do when evergreen hits the timeout (`post:` tasks are run automatically)
+timeout:
+- command: shell.exec
+  params:
+    script: |
+      ls -la
+
+pre_error_fails_task: true
+pre:
+- command: git.get_project
+  params:
+    directory: src
+- command: shell.exec
+  params:
+    working_dir: "src"
+    script: |
+      export PROJECT_DIRECTORY="$(pwd)"
+      export MONGOCRYPT_LIB_DIR="$PROJECT_DIRECTORY/native/${libmongocrypt_os}/lib"
+      export LD_LIBRARY_PATH="$MONGOCRYPT_LIB_DIR:$LD_LIBRARY_PATH"
+
+      cat <<EOT > expansion.yml
+      PREPARE_SHELL: |
+        set -o errexit
+        set -o xtrace
+        export PROJECT_DIRECTORY="$PROJECT_DIRECTORY"
+        export MONGOCRYPT_LIB_DIR="$MONGOCRYPT_LIB_DIR"
+        export LD_LIBRARY_PATH="$LD_LIBRARY_PATH"
+      EOT
+      cat expansion.yml
+- command: expansions.update
+  params:
+    file: src/expansion.yml
+- command: shell.exec
+  params:
+    working_dir: "src"
+    script: |
+      ${PREPARE_SHELL}
+      .evergreen/install-dependencies.sh
+
+tasks:
+- name: "compile-only"
+  commands:
+  - command: shell.exec
+    type: test
+    params:
+      working_dir: "src"
+      script: |
+        ${PREPARE_SHELL}
+        .evergreen/compile-only.sh
+
+- name: "test"
+  commands:
+  - command: shell.exec
+    type: test
+    params:
+      working_dir: "src"
+      script: |
+        ${PREPARE_SHELL}
+        .evergreen/run-tests.sh
+
+buildvariants:
```

Comment:
The Rust `mongodb` and `bson` crates use matrix variants; it turns out that's [deprecated and disrecommended](https://github.com/evergreen-ci/evergreen/wiki/Project-Configuration-Files#matrix-variant-definition), so I didn't use it here.

Even without that, I don't think a full matrix will be useful for this project - compile-only should be tested across platforms, and `mongocrypt-sys` should have tests run on all of them, but if those pass there's no reason to run `mongocrypt` tests on more than a single one.